### PR TITLE
VPN-5628: Increase minimum CMake to 3.18

### PR DESCRIPTION
**Pull request by oskirby**
## Description
The introduction of QML modules in the 2.17 release revealed a bug in the registration of QML modules where resources from those modules would fail to load when built with CMake 3.16. This scenario only occurs on Ubuntu 20.04 (Focal Fossa). To resolve the issue, we need to upgrade the version of CMake available to our build workers.

While this PR looks simple, the real work is in backporting CMake 3.18 to Ubuntu/focal. This was done in the `ppa:okirby/qt6-backports` repository, with the addition of this [package](https://launchpad.net/~okirby/+archive/ubuntu/qt6-backports/+sourcepub/15217683/+listing-archive-extra)

## Reference
Github #8163 ([VPN-5628](https://mozilla-hub.atlassian.net/browse/VPN-5628))
CMake backport: https://github.com/oskirby/qt6-packaging/commit/6c80f268b1987dfe371bdac6158654014638e639

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

[VPN-5628]: https://mozilla-hub.atlassian.net/browse/VPN-5628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,9 +89,6 @@ endif()
 ## Toolchain Setup
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.17)
-    cmake_policy(SET CMP0099 OLD)
-endif()
```
Comment (Gela) [inline CMakeLists.txt:94]: why was this removed?

File: lottie/CMakeLists.txt [hunk 2]
```diff
@@ -14,9 +14,6 @@ set(CMAKE_AUTOUIC ON)
 if(NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     set(CMAKE_FOLDER "Lottie")
 endif()
-if(${CMAKE_VERSION} VERSION_GREATER_EQUAL 3.17)
-    cmake_policy(SET CMP0099 OLD)
-endif()
```
Comment (brizental) [inline lottie/CMakeLists.txt:19]: Why remove this?

**Review by brizental — APPROVED** [review]
r+wc

File: lottie/CMakeLists.txt (hunk 2 above)
Comment (oskirby) [reply lottie/CMakeLists.txt:19]: Gela just asked me the same question, see the answer [here](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8324#discussion_r1364067283)
